### PR TITLE
L1T DQM Offline Stage 2 upgrades

### PR DESCRIPTION
**Pull request by cbbrainerd**
Changed L1T DQM Offline sequences to use L1T Stage 2 modules and to produce the same histograms as L1T DQM Online sequences.

### CODE REVIEW

**Comment by dmitrijus** [comment]
Please do not change anything in DQMServices/Components. You don't really need to modify customDQM file, if you do, make a copy to L1Trigger directory.

**Comment by cbbrainerd** [comment]
@dmitrijus I rolled back the changes to DQMServices/Components and moved the modified customDQM file to DQMOffline/L1Trigger.

File: DQMOffline/Configuration/python/DQMOffline_cff.py [hunk 1]
```diff
@@ -20,6 +20,10 @@
 from DQMOffline.L1Trigger.L1TriggerDqmOffline_cff import *
 from DQM.CTPPS.totemDQM_cff import *
 
+##test
```
Comment (dmitrijus) [inline DQMOffline/Configuration/python/DQMOffline_cff.py]: test?
Perhaps, you forgot to remove it once you have done testing :(

**Comment by dmitrijus** [comment]
One more comment!

**Comment by cbbrainerd** [comment]
@dmitrijus That should do it.

File: DQMOffline/Configuration/python/DQMOffline_cff.py [hunk 2]
```diff
@@ -20,6 +20,8 @@
 from DQMOffline.L1Trigger.L1TriggerDqmOffline_cff import *
 from DQM.CTPPS.totemDQM_cff import *
 
+hcalOfflineSourceSequence.remove(tpTask)
```
Comment (dmitrijus) [inline DQMOffline/Configuration/python/DQMOffline_cff.py]: I doubt HCAL will be too happy to see their sequence being modified.
Get someone from HCAL to clarify - also this should be done via hcalOfflineSourceSequence, not hack-removed via DQMOffline_cff.py

File: DQMOffline/L1Trigger/python/L1TriggerDqmOffline_cff.py [hunk 3]
```diff
@@ -19,7 +19,9 @@
 
 import DQMServices.Components.DQMEnvironment_cfi
 dqmEnvL1T = DQMServices.Components.DQMEnvironment_cfi.dqmEnv.clone()
-dqmEnvL1T.subSystemFolder = 'L1T'
+dqmEnvL1T.subSystemFolder = 'L1T2016'
```
Comment (dmitrijus) [inline DQMOffline/L1Trigger/python/L1TriggerDqmOffline_cff.py]: I don't care about the name of this folder myself, but that will most definitely break something.

**Comment by dmitrijus** [comment]
please test
I wrote a few more comments too.

**Comment by cbbrainerd** [comment]
@dmitrijus That should fix the crashes seen in the tests.

File: DQMOffline/L1Trigger/python/L1TriggerDqmOffline_cff.py (hunk 3 above)
Comment (cbbrainerd) [inline DQMOffline/L1Trigger/python/L1TriggerDqmOffline_cff.py]: I believe we are dropping support for the legacy L1T DQM. Anyway, we were asked to populate the stage 2 L1T histograms into the L1T2016 folder.

**Comment by dmitrijus** [comment]
please test

**Comment by cbbrainerd** [comment]
@dmitrijus Fixed those crashes.

**Comment by cbbrainerd** [comment]
@dmitrijus Should be ready now.

**Comment by dmitrijus** [comment]
please test

**Comment by cbbrainerd** [comment]
@dmitrijus Are there any other problems on our PR?

**Comment by dmitrijus** [comment]
Yes, it lacks L1 signature.

**Comment by dmitrijus** [comment]
+1

**Comment by cbbrainerd** [comment]
@mulhearn Can you check this out when you get a chance?

**Comment by rekovic** [comment]
+1

**Comment by davidlange6** [comment]
Hi @cbbrainerd @rekovic - the isChosen methods need to be modernized as that syntax is no longer allowed. Thanks
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCmsDriverEras

**Comment by kreczko** [comment]
@davidlange6 : out of curiosity, what is the correct syntax?
As I understand, these are not allowed
```python
if stage2L1Trigger.isChosen():
    from DQM.L1TMonitorClient.L1TStage2EmulatorMonitorClient_cff import *
else:
    from DQM.L1TMonitorClient.L1TEMUMonitorClient_cff import *
# or
from Configuration.StandardSequences.Eras import eras
if eras.run2_common.isChosen() :
    myProducer.customisedParameter=cms.string("valueForRun2")
```

**Comment by davidlange6** [comment]
did you have a look at the twiki page that I pointed you to?

**Comment by kreczko** [comment]
@davidlange6 Yes, and it is not clear (I'm a newbie w.r.t eras).
It says `"if" statements (DISALLOWED)`, but I cannot see the alternative.

**Comment by mulhearn** [comment]
@davidlange6  I'm stumped as to why isChosen is now disallowed.  This allowed for nice clean configs that handle the era's in an easily maintainable way.  Without it, the configs immediately become a nightmare of unnecessarily interrelated config file changes.  Why should era A need to know what era B does, so to turn off those settings?  Can the folks behind this policy provide a clean example that handles non-trivial configuration where era A and era B use different, orthogonal modules, such that era A section does not need to be updated if era B adds a new module, and vice versa?  It is entirely possible I'm misreading the twiki and this is still possible with the non-deprecated commands, so this is not simply a rhetorical question.

**Comment by mulhearn** [comment]
At a minimum, the perspective should be wrt the latest era, not legacy.  As is, this requires us to configure for legacy (0.01% use case) and then have gnarly overrides for latest config (99.99% use case).  Seems backwards!

**Comment by mulhearn** [comment]
And how is one supposed to handle conditions?  What is correct procedure to override with a particular dummy ESProducer for Stage1 versus Stage2?  Is there now a toReplaceWith functionality for ESProducers?

**Comment by mulhearn** [comment]
I think for the conditions, this is fatal for us.  I can't see (even an ugly hack) to handle this without using isChosen.

**Comment by mulhearn** [comment]
Can we continue to use (deprecated, but not DISALLOWED) modify functions?
```
from Configuration.StandardSequences.Eras import eras

def _modifyMyModuleForRun2( moduleObject ) :
    for item in moduleObject.parameterList :
        item.value+=9

eras.run2_common.toModify( myProducer, func=_modifyMyModuleForRun2 )
```

**Comment by rekovic** [comment]
hold

**Comment by rekovic** [comment]
@cbbrainerd It looks like you picked up many unnecessary changes.   Please revert the last commit in your branch
<pre>
git revert c2b397b
</pre>

and push it again, to restart.

**Comment by cbbrainerd** [comment]
@rekovic Sorry about that. That should do it.

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
unhold

**Comment by dmitrijus** [comment]
+1

**Comment by silviodonato** [comment]
Can someone close this PR?

**Pull request closed without merging**